File: app/src/main/java/de/build_a_hero/app/HttpGetRequest.java
package de.build_a_hero.app;

import android.os.AsyncTask;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.Buffer;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

/**
 * Created by ninad on 15.01.2018.
 */

public class HttpGetRequest extends AsyncTask<String, Void, String> {

    public static final String REQUEST_METHOD = "GET";
    public static final int READ_TIMEOUT = 15000;
    public static final int CONNECTION_TIMEOUT = 15000;



    @Override
    protected String doInBackground(String... params) {
        String url = params[0];
        String result;
        String inputLine;



        try{
            URL myUrl = new URL(url);

            HttpURLConnection connection = (HttpURLConnection) myUrl.openConnection();

            connection.setRequestMethod(REQUEST_METHOD);
            connection.setReadTimeout(READ_TIMEOUT);
            connection.setConnectTimeout(CONNECTION_TIMEOUT);

            connection.connect();

            InputStreamReader streamReader = new InputStreamReader(connection.getInputStream());

            BufferedReader reader = new BufferedReader(streamReader);
            StringBuilder stringBuilder = new StringBuilder();


            while((inputLine = reader.readLine()) != null){
                stringBuilder.append(inputLine + "\n");
            }

            reader.close();
            streamReader.close();

            result = stringBuilder.toString();

        }catch(IOException e) {
            e.printStackTrace();
            result = null;
        }
        return result;
    }

    @Override
    protected void onPostExecute(String result){
        super.onPostExecute(result);
    }

}

File: app/src/main/java/de/build_a_hero/app/HttpGetRequest.kt
package de.build_a_hero.app

import android.os.AsyncTask

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class HttpGetRequest : AsyncTask<String, Void, String>() {


    override fun doInBackground(vararg params: String): String? {
        val url = params[0]
        var result: String?
        var inputLine: String


        try {
            val myUrl = URL(url)

            val connection = myUrl.openConnection() as HttpURLConnection

            connection.requestMethod = REQUEST_METHOD
            connection.readTimeout = READ_TIMEOUT
            connection.connectTimeout = CONNECTION_TIMEOUT

            connection.connect()

            val streamReader = InputStreamReader(connection.inputStream)

            val reader = BufferedReader(streamReader)
            val stringBuilder = StringBuilder()


            while (true) {
                inputLine = reader.readLine() ?: break

                stringBuilder.append(inputLine + "\n")

            }

            reader.close()
            streamReader.close()

            result = stringBuilder.toString()

        } catch (e: IOException) {
            e.printStackTrace()
            result = null
        }

        return result
    }

    companion object {

        val REQUEST_METHOD = "GET"
        val READ_TIMEOUT = 15000
        val CONNECTION_TIMEOUT = 15000
    }

}

